### PR TITLE
Bug in the vector rotation operator

### PR DESCRIPTION
**Pull request by simone-silvestri**
The meridional velocity had the wrong sign in the vector rotation operator for switching from extrinsic to intrinsic coordinates in an `OrthogonalSphericalShellGrid`.
The test was also poorly designed (my bad) and conspired to hide the bug. This PR should fix the rotation and update the test to make sure everything is correct.

### CODE REVIEW

**Comment by glwagner** [comment]
Can we add a test that uses `OrthogonalSphericalShellGrids`?

File: test/test_cubed_spheres.jl [hunk 1]
```diff
@@ -20,6 +22,116 @@ using Oceananigans.Models.HydrostaticFreeSurfaceModels: VerticalVorticityField
...
+function kinetic_energy(u, v)
+    ke = Field(0.5 * (u * u + v * v))
+    return compute!(ke)
+end
+
+function test_vector_rotation(grid)
+    u = XFaceField(grid)
+    v = YFaceField(grid)
+    
+    # Purely zonal flow.
```
Comment (glwagner) [inline test/test_cubed_spheres.jl:34]: not for all grids right?

**Comment by glwagner** [comment]
Perhaps we don't have this, but can we rotate a `LatitudeLongitudeGrid` by a certain angle, and then test that the vectors are rotated exactly by that angle? This would provide a simple deterministic test. This could work for a very small grid on the equator, for example...

**Review by glwagner — APPROVED** [review]
I see that the challenge of writing tests is that we don't have a good way of generating general grids...

File: test/test_ensemble_hydrostatic_free_surface_models.jl [hunk 2]
```diff
@@ -4,6 +4,41 @@ using Oceananigans.Models.HydrostaticFreeSurfaceModels: ColumnEnsembleSize, Slic
...
+    run!(sic_simulation)
+    run!(per_simulation)
+    
+    @info "Testing Single column grid results..."
+    
+    @test all(sic_model.velocities.u.data[i, j, :] .≈ per_model[i, j].velocities.u.data[1, 1, :])
+    @test all(sic_model.velocities.v.data[i, j, :] .≈ per_model[i, j].velocities.v.data[1, 1, :])
+    @test all(sic_model.tracers.c.data[i, j, :]    .≈ per_model[i, j].tracers.c.data[1, 1, :])
+end
+
```
Comment (glwagner) [inline test/test_ensemble_hydrostatic_free_surface_models.jl]: was this sent to the wrong PR?

Comment (simone-silvestri) [reply test/test_ensemble_hydrostatic_free_surface_models.jl]: yeah this was meant to be in #3741. I have moved it there

File: test/test_cubed_spheres.jl (hunk 1 above)
Comment (simone-silvestri) [reply test/test_cubed_spheres.jl]: right, it's in the extrinsic frame. I have changed the comment

**Pull request merged**